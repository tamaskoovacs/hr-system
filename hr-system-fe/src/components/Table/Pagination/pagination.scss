@import 'styles/colors';

.pagination-container {
    display: flex;
    list-style-type: none;
    justify-content: center;
  
    .pagination-item {
      padding: 0 12px;
      height: 32px;
      text-align: center;
      margin: auto 4px;
      color: $black;
      display: flex;
      box-sizing: border-box;
      align-items: center;
      letter-spacing: 0.01071em;
      border-radius: 16px;
      line-height: 1.43;
      font-size: 13px;
      min-width: 32px;
  
      &.dots:hover {
        background-color: transparent;
        cursor: default;
      }
      &:hover {
        background-color: $grey;
        cursor: pointer;
      }
  
      &.selected {
        background-color: $grey;
      }
  
      .arrow {
        &::before {
          position: relative;
          content: '';
          display: inline-block;
          width: 0.4em;
          height: 0.4em;
          border-right: 0.12em solid $black;
          border-top: 0.12em solid $black;
        }
  
        &.left {
          transform: rotate(-135deg) translate(-50%);
        }
  
        &.right {
          transform: rotate(45deg);
        }
      }
  
      &.disabled {
        pointer-events: none;
  
        .arrow::before {
          border-right: 0.12em solid $black;
          border-top: 0.12em solid $black;
        }
  
        &:hover {
          background-color: transparent;
          cursor: default;
        }
      }
    }
  }
  
  