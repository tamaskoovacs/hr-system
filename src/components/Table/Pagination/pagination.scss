@import 'styles/colors';

.pagination-container {
  display: flex;
  list-style-type: none;
  justify-content: center;

  .pagination-item {
    padding: 0 12px;
    height: 32px;
    text-align: center;
    margin: auto 4px;
    color: $table-pagination-border;
    display: flex;
    box-sizing: border-box;
    align-items: center;
    letter-spacing: 0.01071em;
    border-radius: 16px;
    line-height: 1.43;
    font-size: 13px;
    min-width: 32px;

    &.dots:hover {
      background-color: transparent;
      cursor: default;
    }
    &:hover {
      background-color: $table-pagination-background;
      cursor: pointer;
    }

    &.selected {
      background-color: $table-pagination-background;
    }

    .arrow {
      &::before {
        position: relative;
        content: '';
        display: inline-block;
        width: 0.4em;
        height: 0.4em;
        border-right: 0.12em solid $table-pagination-border;
        border-top: 0.12em solid $table-pagination-border;
      }

      &.left {
        transform: rotate(-135deg) translate(-50%);
      }

      &.right {
        transform: rotate(45deg);
      }
    }

    &.disabled {
      pointer-events: none;

      .arrow::before {
        border-right: 0.12em solid $table-pagination-border;
        border-top: 0.12em solid $table-pagination-border;
      }

      &:hover {
        background-color: transparent;
        cursor: default;
      }
    }
  }
}
