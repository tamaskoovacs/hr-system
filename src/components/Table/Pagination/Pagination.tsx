import React from 'react';
import classnames from 'classnames';
import { usePagination, DOTS } from 'hooks/usePagination';
import './pagination.scss';
import { PaginationProps } from 'interfaces/interfaces';

const Pagination: React.FC<PaginationProps> = ({
  onPageChange,
  totalCount,
  siblingCount = 1,
  currentPage,
  pageSize,
  className
}) => {
  const paginationRange = usePagination({ currentPage, totalCount, siblingCount, pageSize });

  if (currentPage === 0 || (paginationRange?.length ?? 0) < 2) {
    return null;
  }

  const onNext = () => {
    onPageChange(currentPage + 1);
  };

  const onPrevious = () => {
    onPageChange(currentPage - 1);
  };

  let lastPage = paginationRange?.[paginationRange?.length - 1];
  return (
    <ul className={classnames('pagination-container', { [className]: className })}>
      <li
        className={classnames('pagination-item', { disabled: currentPage === 1 })}
        onClick={onPrevious}
        data-testid="left-arrow-li"
      >
        <div className="arrow left" />
      </li>
      {paginationRange?.map((pageNumber, index) => {
        if (pageNumber === DOTS) {
          return (
            <li className="pagination-item dots" key={index}>
              &#8230;
            </li>
          );
        }

        return (
          <li
            className={classnames('pagination-item', {
              selected: pageNumber === currentPage
            })}
            onClick={() => onPageChange(pageNumber)}
            key={index}
            data-testid={`pagination-li-${index + 1}`}
          >
            {pageNumber}
          </li>
        );
      })}
      <li
        className={classnames('pagination-item', {
          disabled: currentPage === lastPage
        })}
        onClick={onNext}
        data-testid="right-arrow-li"
      >
        <div className="arrow right" />
      </li>
    </ul>
  );
};

export default Pagination;
