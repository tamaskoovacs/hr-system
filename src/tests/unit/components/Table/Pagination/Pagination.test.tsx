import React from 'react';
import { render, screen } from '@testing-library/react';
import Pagination from 'components/Table/Pagination/Pagination';

describe('Pagination component', () => {
  test('It should matches snapshot', () => {
    const view = render(
      <Pagination className={'pagination-bar'} currentPage={1} totalCount={26} pageSize={25} onPageChange={jest.fn()} />
    );
    expect(view.container).toMatchSnapshot();
  });

  test('It should appear the Pagination component when the totalCount is greater than the pageSize', () => {
    render(<Pagination className={'pagination-bar'} currentPage={1} totalCount={26} pageSize={25} onPageChange={jest.fn()} />);
    const paginationLiElements = screen?.getAllByRole('listitem');
    expect(paginationLiElements)?.toHaveLength(4);
  });

  test('It should not appear the Pagination component when the totalCount is less than the pageSize', () => {
    render(<Pagination className={'pagination-bar'} currentPage={1} totalCount={24} pageSize={25} onPageChange={jest.fn()} />);
    const paginationUlElement = screen?.queryByRole('list');
    expect(paginationUlElement).toBeNull();
  });
});
