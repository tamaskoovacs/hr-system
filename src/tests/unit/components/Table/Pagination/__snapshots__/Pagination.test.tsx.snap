// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component It should matches snapshot 1`] = `
<div>
  <ul
    class="pagination-container pagination-bar"
  >
    <li
      class="pagination-item disabled"
      data-testid="left-arrow-li"
    >
      <div
        class="arrow left"
      />
    </li>
    <li
      class="pagination-item selected"
      data-testid="pagination-li-1"
    >
      1
    </li>
    <li
      class="pagination-item"
      data-testid="pagination-li-2"
    >
      2
    </li>
    <li
      class="pagination-item"
      data-testid="right-arrow-li"
    >
      <div
        class="arrow right"
      />
    </li>
  </ul>
</div>
`;
